python/pyspark/mllib/stat/_statistics.py

```diff
@@ -168,12 +190,34 @@ def corr(x, y=None, method=None):
             raise TypeError("Use 'method=' to specify method name.")
 
         if not y:
-            return callMLlibFunc("corr", x.map(_convert_to_vector), method).toArray()
+            return cast(
+                JavaObject, callMLlibFunc("corr", x.map(_convert_to_vector), method)
+            ).toArray()
         else:
-            return callMLlibFunc("corr", x.map(float), y.map(float), method)
+            return cast(
+                Union[float, Matrix],
```

Comment:
That's not `Union`. If you are in this specific branch, you have two `RDD[float]` and result should be float.

Reply:
@nchammas Just looking at this, we might have to make adjust signatures for `callMLlibFunc` and `callMLlibFunc` which match `_java2py` but don't take into account types that are automatically converted to Python types.